File: frontend/src/types/job.ts
export type JobStatus =
  | 'QUEUED'
  | 'PARSING'
  | 'NORMALIZING'
  | 'MATCHING'
  | 'AI_PREDICTING'
  | 'VALIDATING'
  | 'REVIEW_REQUIRED'
  | 'READY'
  | 'EXPORTED'
  | 'FAILED'

export interface StatusHistoryEntry {
  status: JobStatus
  timestamp: string
}

export interface ProcessingJob {
  id: string
  uploaded_file_id: string
  status: JobStatus
  status_history: StatusHistoryEntry[]
  total_transactions: number
  auto_matched_count: number
  ai_predicted_count: number
  manual_review_count: number
  export_ready_count: number
  error_message: string | null
  created_at: string
}

export interface ProcessingJobListResponse {
  items: ProcessingJob[]
  total: number
  page: number
  page_size: number
}

export interface UploadResponse {
  job_id: string
}

export const TERMINAL_JOB_STATUSES: JobStatus[] = ['READY', 'REVIEW_REQUIRED', 'FAILED', 'EXPORTED']

File: backend/app/schemas/job.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.processing_job import JobStatus


class StatusHistoryEntry(BaseModel):
    status: JobStatus
    timestamp: datetime


class ProcessingJobOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    uploaded_file_id: uuid.UUID
    status: JobStatus
    status_history: list[StatusHistoryEntry]
    total_transactions: int
    auto_matched_count: int
    ai_predicted_count: int
    manual_review_count: int
    export_ready_count: int
    error_message: str | None
    created_at: datetime


class UploadResponse(BaseModel):
    job_id: uuid.UUID


class ProcessingJobListResponse(BaseModel):
    items: list[ProcessingJobOut]
    total: int
    page: int
    page_size: int
